webrob/pages/admin/db.py

```diff
@@ -145,3 +160,80 @@ def db_remove_route(table):
     if data is not None and 'id' in data and db_find(cls, data['id']):
         db_remove(cls, data['id'])
     return jsonify(result=None)
+
+
+@app.route('/edit_neem_hub_settings')
+@admin_required
+def render_neem_hub_settings_page_get():
+    # PasswordForm used for validating given password field
+    form = PasswordForm()
+    try:
+        neemHubSettings = get_settings()
+    except SQLAlchemyError as e:
+        raise SQLAlchemyErrorException('while connecting to sql db raises an exception', exc=e)
+    except NoResultFound as e:
+        raise SQLAlchemyErrorException('while connecting to sql db returns no result found', exc=e)
+
+    return render_template('admin/neem_hub_settings.html', form=form, neemHubSettings=neemHubSettings)
+
+
+@app.route('/save_edited_neem_hub_settings', methods=['POST'])
+@admin_required
+def render_neem_hub_settings_post():
+    app.logger.debug('render neem hub settings post method.... ')
+    req = request.form
+    if req is not None:
+        neemHubSettings = NEEMHubSettings()
+
+        # first find if there is any existing entry in db
+        count = get_settings_count()
+        if count > 0:
+            neemHubSettings = get_settings()
+
+        neemHubSettings.MONGO_HOST = req.get("MONGO_HOST")
+        neemHubSettings.MONGO_PORT = req.get("MONGO_PORT")
+        neemHubSettings.MONGO_USER = req.get("MONGO_USER")
+        neemHubSettings.MONGO_DB = req.get("MONGO_DB")
+        neemHubSettings.MONGO_PASS = req.get("MONGO_PASS")
+
+        db.session.add(neemHubSettings)
+        db.session.commit()
+        app.logger.info('Configuration has been stored!')
+
+        try:
+            get_mongo_db_meta_collection()
+            # if connection is secured then update neem_ids from mongodb meta collection so that neem discovery page has latest updates
+            neem_manager.set_neem_ids()
+            app.logger.debug('------------ neem manager updated ------------')
+
+        except ConnectionFailure as e:
```

Comment:
why not adding a flask handler for this error?

Reply:
here we do handle an error and points to flask handler.

Reply:
done. Now ConnectionFailure has its own flask handler.